File: KoRadio/KoRadio.Services/Database/CompanyEmployee.cs
using System;
using System.Collections.Generic;

namespace KoRadio.Services.Database;

public partial class CompanyEmployee:ISoftDelete
{
    public int UserId { get; set; }

    public int CompanyId { get; set; }

    public bool IsDeleted { get; set; }

    public bool IsApplicant { get; set; }

    public int? CompanyRoleId { get; set; }

    public DateTime DateJoined { get; set; }

    public int CompanyEmployeeId { get; set; }

    public virtual Company Company { get; set; } = null!;

    public virtual ICollection<CompanyJobAssignment> CompanyJobAssignments { get; set; } = new List<CompanyJobAssignment>();

    public virtual CompanyRole? CompanyRole { get; set; }

    public virtual ICollection<EmployeeTask> EmployeeTasks { get; set; } = new List<EmployeeTask>();

    public virtual User User { get; set; } = null!;
}

File: KoRadio/UI/ko_radio_desktop/lib/models/company_employee.dart

import 'package:json_annotation/json_annotation.dart';
import 'package:ko_radio_desktop/models/company.dart';
import 'package:ko_radio_desktop/models/company_role.dart';
import 'package:ko_radio_desktop/models/user.dart';


part 'company_employee.g.dart';
@JsonSerializable()
class CompanyEmployee {
  int companyEmployeeId;
  int? userId;
  int? companyId;
  bool? isDeleted;
  bool? isApplicant;
  String? companyName;
  int? companyRoleId;
  String? companyRoleName;
  DateTime? dateJoined;
  User? user;


  CompanyEmployee( this.companyEmployeeId, this.userId, this.companyId, this.isDeleted, this.isApplicant, this.user);

  factory CompanyEmployee.fromJson(Map<String, dynamic> json) => _$CompanyEmployeeFromJson(json);
  Map<String, dynamic> toJson() => _$CompanyEmployeeToJson(this);
}